packages/next/src/build/swc/options.ts

```diff
@@ -211,6 +211,8 @@ function getBaseSWCOptions({
     // For app router we prefer to bundle ESM,
     // On server side of pages router we prefer CJS.
     preferEsm: esm,
+
+    decoratorVersion: compilerOptions?.decoratorVersion ?? 'legacy',
```

Comment:
Alternatively we can branch from js and use https://swc.rs/docs/configuration/compilation#jsctransformdecoratorversion instead